### PR TITLE
Add Release Api

### PR DESCRIPTION
**Pull request by eutkin**
https://docs.gitlab.com/ee/api/tags.html#create-a-new-release
https://docs.gitlab.com/ee/api/tags.html#update-a-release

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/CommitsApi.java [hunk 1]
```diff
@@ -1,21 +1,18 @@
 package org.gitlab4j.api;
 
+import org.gitlab4j.api.models.*;
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/CommitsApi.java]: It is good practice to not use wildcard imports, please change this to only import the types being used.

File: src/main/java/org/gitlab4j/api/RepositoryApi.java [hunk 2]
```diff
@@ -1,5 +1,13 @@
 package org.gitlab4j.api;
 
+import org.gitlab4j.api.GitLabApi.ApiVersion;
+import org.gitlab4j.api.models.*;
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/RepositoryApi.java]: It is good practice to not use wildcard imports, please change this to only import the types being used.

File: src/test/java/org/gitlab4j/api/TestCommitsApi.java [hunk 3]
```diff
@@ -1,26 +1,20 @@
...
 import org.junit.BeforeClass;
 import org.junit.FixMethodOrder;
 import org.junit.Test;
 import org.junit.runners.MethodSorters;
 
+import javax.ws.rs.core.Response;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.Assert.*;
```
Comment (gmessner) [inline src/test/java/org/gitlab4j/api/TestCommitsApi.java]: It is good practice to not use wildcard imports, please change this to only import the types being used.

File: src/test/java/org/gitlab4j/api/TestCommitsApi.java [hunk 4]
```diff
@@ -1,26 +1,20 @@
...
-import java.util.List;
-
-import javax.ws.rs.core.Response;
-
 import org.gitlab4j.api.GitLabApi.ApiVersion;
-import org.gitlab4j.api.models.Comment;
-import org.gitlab4j.api.models.Commit;
-import org.gitlab4j.api.models.Diff;
-import org.gitlab4j.api.models.Project;
+import org.gitlab4j.api.models.*;
```
Comment (gmessner) [inline src/test/java/org/gitlab4j/api/TestCommitsApi.java]: It is good practice to not use wildcard imports, please change this to only import the types being used.

File: src/main/java/org/gitlab4j/api/models/CommitRef.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+@XmlAccessorType(XmlAccessType.FIELD)
+public class CommitRef {
+
+    private RefType type;
+    private String name;
+
+   public enum RefType {
+        branch, tag, all
+    }
+
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/CommitRef.java]: All enums should be uppercase, and to make the enum work correctly with this library this enum should look like this:
```
public enum RefType {

    BRANCH, TAG, ALL;

    private static JacksonJsonEnumHelper<RefType> enumHelper = new JacksonJsonEnumHelper<>(RefType.class);

    @JsonCreator
    public static RefType forValue(String value) {
        return enumHelper.forValue(value);
    }

    @JsonValue
    public String toValue() {
        return (enumHelper.toString(this));
    }

    @Override
    public String toString() {
        return (enumHelper.toString(this));
    }
}
```

**Comment by eutkin** [comment]
ok, i done, thank you

**Comment by gmessner** [comment]
@eutkin 
This has been released in 4.8.28

**Comment by gmessner** [comment]
@eutkin Just a heads up.

The createRelease() and updateRelease() methods along with all the tag related methods have been moved to a new API called TagsApi.  This was done as this library tries to maintain a one-to-one relationship with the GitLab API docs, and these methods are all in the Tags API at:
https://docs.gitlab.com/ce/api/tags.html

Thanks again for your contribution.

**Pull request merged**